sdk/ai/azure-ai-projects/azure/ai/projects/aio/operations/_patch.py

```diff
@@ -1834,37 +2034,47 @@ async def submit_tool_outputs_to_stream(
 
     @overload
     async def submit_tool_outputs_to_stream(
-        self, thread_id: str, run_id: str, body: IO[bytes], *, content_type: str = "application/json", **kwargs: Any
-    ) -> _models.AsyncAgentRunStream:
+        self,
+        thread_id: str,
+        run_id: str,
+        *,
+        tool_outputs: List[_models.ToolOutput],
+        content_type: str = "application/json",
+        event_handler: _models.BaseAsyncAgentEventHandlerT,
+        **kwargs: Any,
+    ) -> _models.AsyncAgentRunStream[_models.BaseAsyncAgentEventHandlerT]:
         """Submits outputs from tools as requested by tool calls in a stream. Runs that need submitted tool
         outputs will have a status of 'requires_action' with a required_action.type of
-        'submit_tool_outputs'.
+        'submit_tool_outputs'.  terminating when the Run enters a terminal state with a ``data: [DONE]`` message.
 
         :param thread_id: Required.
         :type thread_id: str
         :param run_id: Required.
         :type run_id: str
-        :param body: Required.
-        :type body: IO[bytes]
-        :keyword content_type: Body Parameter content-type. Content type parameter for binary body.
+        :keyword tool_outputs: Required.
+        :paramtype tool_outputs: list[~azure.ai.projects.models.ToolOutput]
+        :keyword content_type: Body Parameter content-type. Content type parameter for JSON body.
          Default value is "application/json".
         :paramtype content_type: str
+        :keyword event_handler: The event handler to use for processing events during the run. Default
+            value is None.
+        :paramtype event_handler: ~azure.ai.projects.models.AsyncAgentEventHandler
         :return: AgentRunStream.  AgentRunStream is compatible with Iterable and supports streaming.
         :rtype: ~azure.ai.projects.models.AsyncAgentRunStream
         :raises ~azure.core.exceptions.HttpResponseError:
         """
 
     @distributed_trace_async
-    async def submit_tool_outputs_to_stream(
+    async def submit_tool_outputs_to_stream(  # pyright: ignore[reportInconsistentOverload]
```

Comment:
Why do we need to type ignore inconsistent overloads here? I'd expect that the implementation signature correctly should eliminate the need for `typing.cast` below.

Reply:
I guess it is because of generic..   The code below requires casting.

```
        if not event_handler:
            event_handler = cast(_models.BaseAsyncAgentEventHandlerT, _models.AsyncAgentEventHandler())
 ```
 
But this is fine.
```
        a: Optional[_models.BaseAsyncAgentEventHandlerT] = event_handler
```